tests/foreman/ui/test_user.py

```diff
@@ -1316,6 +1322,17 @@ def setUpClass(cls):  # noqa
         ).create()
         cls.ldap_server_name = authsource_attrs.name
 
+    def check_external_user(self):
+        """Check whether external user is active and reachable. That
+        operation also add that user into application system for internal
+        configuration procedures
+        """
+        strategy, value = locators['login.loggedin']
```

Comment:
We do not use `strategy, value` anymore
```
with Session(self, self.ldap_user_name, self.ldap_user_passwd):
self.assertIsNotNone(self.login.wait_until_element(locators['login.loggedin'] % self.ldap_user_name))
```

Reply:
Okay, here is a very important information
By fetching and in fact adding external user into Satellite you making that user affected by any test case where you will use it. That mean not only current class or even not current module will be affected, but any code piece that will use it, so you have to run all cases in one thread:
```
@run_in_one_thread
class ActiveDirectoryUserTestCase(UITestCase):
```
and also you have to make sure that user deleted from the system after each test case, to clear all interactions that you performed with that user (e.g. added roles):
```
    def tearDown(self):
        with Session(self) as session:
            set_context(session, org=ANY_CONTEXT['org'])
            if self.user.search(self.ldap_user_name):
                self.user.delete(self.ldap_user_name)
        super(ActiveDirectoryUserTestCase, self).tearDown()
```